server.coffee

```diff
@@ -299,6 +299,19 @@ render_template_with_context = (ctx) ->
       resolve ctx
   )["catch"]((err) -> ctx.error = err; Promise.reject ctx)
 
+inject_template_name = (ctx) ->
+  new Promise( (resolve, reject) ->
+      ###
+      if ctx.rendered_template.indexOf('EXEC sp_executesql') != -1
+        log.debug "******** Parameterized query detected do something smart here *************"
+      else
+      ###
+      arr = ctx.rendered_template.split('\n')
+      arr.unshift('-- ' + ctx.template_name)
+      ctx.rendered_template = arr.join('\n')
```

Comment:
I was thinking moving forward that if we want to inject the template name in the sp_execute templates it would be easier to find the right place to do the injection this way..We can totally just go it as a string concat for now.